Hunting Queries/DnsEvents/DNS_HighNXDomainCount.txt

```diff
@@ -1,24 +1,82 @@
+| extend truncatedDomain = iff((strlen(Name) - indexof(Name, tostring(split(Name, ".")[-2])) ) >= 7, strcat(tostring(split(Name, ".")[-2]), ".", tostring(split(Name, ".")[-1])) , strcat(tostring(split(Name, ".")[-3]), ".", tostring(split(Name, ".")[-2]), ".", tostring(split(Name, ".")[-1])))
+| where mytld !in ("arris","ati","virtusa","unknowndomain","onion","corp","domain","local","localdomain","host","home","gateway","lan","services","hub","domain.name","WirelessAP","Digicom-ADSL","OpenDNS","dlinkrouter","Dlink","ASUS","device","router","Belkin","DHCP","Cisco");
+//
+let quartileFunction = view (mypercentile:long, startTimeSpan:timespan, endTimeSpan:timespan) {
+(nxDomainDnsEvents
+| where TimeGeneratedUtc between (ago(startTimeSpan)..ago(endTimeSpan))
+| extend truncatedDomain = iff((strlen(Name) - indexof(Name, tostring(split(Name, ".")[-2])) ) >= 7, strcat(tostring(split(Name, ".")[-2]), ".", tostring(split(Name, ".")[-1])) , strcat(tostring(split(Name, ".")[-3]), ".", tostring(split(Name, ".")[-2]), ".", tostring(split(Name, ".")[-1])))
+| summarize domainCount = dcount(truncatedDomain) by ClientIP, bin(TimeGeneratedUtc, 1d)
+| project SearchList = (domainCount), ClientIP
+| summarize qPercentiles = percentiles(SearchList, mypercentile) by ClientIP);
+};
+let firstQT = quartileFunction(25, 7d, 2d) | project-rename percentile_SearchList_25 = qPercentiles;
+let thirdQT = quartileFunction(75, 7d, 2d) | project-rename percentile_SearchList_75 = qPercentiles;
+let threshold = (firstQT
+| join thirdQT on ClientIP
+| extend IPthreshold = percentile_SearchList_75 + (1.5*exp(3)*(percentile_SearchList_75 - percentile_SearchList_25))
+| project ClientIP, IPthreshold);
+let Tab1 = (
+nxDomainDnsEvents
+| where TimeGeneratedUtc > ago(1d)
+| summarize TotalNXLookups=dcount(truncatedDomain) by ClientIP
+| sort by TotalNXLookups desc
+| join ['threshold'] on ClientIP
+| where TotalNXLookups > IPthreshold // Comment this line in order to view results filtered by Global Threshold only. 
+| join kind = leftouter (nxDomainDnsEvents
+    | where TimeGeneratedUtc > ago(1d)
+    | extend SearchList = iff((strlen(Name) - indexof(Name, tostring(split(Name, ".")[-2])) ) >= 7, strcat(tostring(split(Name, ".")[-2]), ".", tostring(split(Name, ".")[-1])) , strcat(tostring(split(Name, ".")[-3]), ".", tostring(split(Name, ".")[-2]), ".", tostring(split(Name, ".")[-1])))
+    | summarize domainCount=count() by SearchList, ClientIP
+    | sort by domainCount desc
+    | project SearchList = strcat(SearchList," (",tostring(domainCount),")"), ClientIP
+    ) on ClientIP
+| summarize SLDs_TotalLookups = makelist(SearchList) by ClientIP, TotalNXLookups, IPthreshold
+| sort by TotalNXLookups desc);
+//
+let quartileFunction2 = view (mypercentile:long, startTimeSpan:timespan) {
+(nxDomainDnsEvents
+| where TimeGeneratedUtc > ago(startTimeSpan)
+| summarize domainCount = dcount(truncatedDomain) by ClientIP, bin(TimeGeneratedUtc, startTimeSpan)
```

Comment:
>ago(startTimeSpan) and bin(TimeGeneratedUtc, startTimeSpan)
That will result in a single bin - is that intended?

Reply:
Yep - I am determining the Global threshold value over the last day, and hence need a single bin

Reply:
in that case can't u just do 
| summarize domainCount=dcount(truncatedDomain) by ClientIP ?
? :)

Reply:
Can do - Will edit